pydra/engine/task.py

```diff
@@ -506,3 +498,104 @@ def _run_task(self):
 
     def _list_outputs(self):
         return list(self.output_)
+
+
+class ContainerTask(ShellCommandTask):
+
+    def __init__(self, input_spec: ty.Optional[ContainerSpec]=None,
+                 output_spec: ty.Optional[ShellOutSpec]=None,
+                 audit_flags: AuditFlag=AuditFlag.NONE,
+                 messengers=None, messenger_args=None, **kwargs):
+
+        if input_spec is None:
+            input_spec = dc.make_dataclass('Inputs', [ARGS], bases=(ContainerSpec,))
+        super(ContainerTask, self).__init__(input_spec=input_spec,
+                                            audit_flags=audit_flags,
+                                            messengers=messengers,
+                                            messenger_args=messenger_args,
+                                            **kwargs)
+
+    @property
+    def cmdline(self):
+        return ' '.join(self.container_args + self.command_args)
+
+    @property
+    def container_args(self):
+        if self.inputs.container is None:
+            raise AttributeError('Container software is not specified')
+        cargs = [self.inputs.container, 'run']
+        if self.inputs.container_xargs is not None:
+            cargs.extend(self.inputs.container_xargs)
+        if self.inputs.image is None:
+            raise AttributeError('Container image is not specified')
+        cargs.append(self.inputs.image)
+        return cargs
+
+    def binds(self, opt):
+        """Specify mounts to bind from local filesystems to container
+
+        `bindings` are tuples of (local path, container path, bind mode)
+        """
+        bargs = []
+        for binding in self.inputs.bindings:
+            lpath, cpath, mode = binding
+            if mode is None:
+                mode = 'rw'  # default
+            bargs.extend([opt, '{0}:{1}:{2}'.format(lpath, cpath, mode)])
+        return bargs
+
+    def _run_task(self):
+        self.output_ = None
+        args = self.container_args + self.command_args
+        if args:
+            self.output_ = execute(args)
+
+
+class DockerTask(ContainerTask):
```

Comment:
perhaps we can use a function instead of a class for this, and use the function inside the class above.